src/pages/api/identity.tsx

```diff
@@ -0,0 +1,44 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { addIdentityToUser } from "../../command/addIdentityToUser";
+import { removeIdentityFromUser } from "../../command/removeIdentityFromUser";
+
+export default async (
+  request: NextApiRequest,
+  response: NextApiResponse,
+): Promise<void> => {
+  const { userId, type, value } = request.body;
+
+  if (request.method === "POST") {
+    return await addIdentityToUser({ userId, type, value })
+      .then(() => {
+        response.statusCode = 200;
+
+        return response.json("succeed");
+      })
+      .catch((error) => {
+        console.log(error);
+
+        response.statusCode = 400;
+
+        return response.json("failed");
+      });
+  } else if (request.method === "DELETE") {
+    return await removeIdentityFromUser({ userId, type, value })
+      .then(() => {
+        response.statusCode = 200;
+
+        return response.json("succeed");
+      })
+      .catch((error) => {
+        console.log(error);
+
+        response.statusCode = 400;
+
+        return response.json("failed");
+      });
+  }
+  response.statusCode = 400;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405

And can we do something like `response.status(405).end()`?

Reply:
Good error code to know. I can't use status as a method though, it's typed as number.

```ts
    class ServerResponse extends OutgoingMessage {
        statusCode: number;
        statusMessage: string;

        constructor(req: IncomingMessage);

        assignSocket(socket: Socket): void;
        detachSocket(socket: Socket): void;
        // https://github.com/nodejs/node/blob/master/test/parallel/test-http-write-callbacks.js#L53
        // no args in writeContinue callback
        writeContinue(callback?: () => void): void;
        writeHead(statusCode: number, reasonPhrase?: string, headers?: OutgoingHttpHeaders): this;
        writeHead(statusCode: number, headers?: OutgoingHttpHeaders): this;
        writeProcessing(): void;
    }
```

Reply:
So I'll just change the error code (the error message has been changed in the next pr, so I won't change it here)